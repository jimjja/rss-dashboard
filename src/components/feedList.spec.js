import React from 'react';
import renderer from 'react-test-renderer';
import FeedList from './feedList';

describe('Feed List component', () => {
  it('should match snapshot', () => {
    const now = new Date();
    const props = {
      title: 'My feed title',
      items: [
        {
          pubDate: now.toString(),
          link: 'http://google.co.uk',
          content: '<h1>I am HTML in string </h1>',
          title: 'Example title',
        },
        {
          pubDate: now.toString(),
          link: 'http://youtube.com',
          content: 'I am text content',
          title: 'RSS Item title',
        },
      ],
    };
    const tree = renderer.create(<FeedList {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render items not found message when no items are passed ', () => {
    const props = {
      title: 'My feed title',
      items: [],
    };
    const tree = renderer.create(<FeedList {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
