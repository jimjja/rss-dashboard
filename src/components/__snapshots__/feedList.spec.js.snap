// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed List component should match snapshot 1`] = `
<div
  style={
    Object {
      "border": "1px solid #ebedf0",
      "borderRadius": 2,
      "padding": "20px",
      "position": "relative",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "border": "1px solid #ebedf0",
        "borderLeft": 0,
        "borderRight": 0,
        "borderTop": 0,
      }
    }
  >
    <h2>
      My feed title
    </h2>
  </div>
  <div
    style={
      Object {
        "border": "1px solid #ebedf0",
        "borderLeft": 0,
        "borderRight": 0,
        "borderTop": 0,
        "color": "#000000a6",
        "padding": "15px",
      }
    }
  >
    <a
      href="http://google.co.uk"
      style={
        Object {
          "color": "#000000a6",
          "fontSize": 18,
          "fontWeight": "bold",
          "textDecoration": "none",
        }
      }
    >
      Example title
    </a>
    <div
      style={
        Object {
          "margin": 0,
        }
      }
    >
      <p
        style={
          Object {
            "display": "inline-block",
            "fontSize": 12,
            "margin": 0,
          }
        }
      >
        2019-2-24
      </p>
    </div>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<h1>I am HTML in string </h1>",
        }
      }
      style={
        Object {
          "fontSize": 12,
          "margin": "1em 0",
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "border": "1px solid #ebedf0",
        "borderLeft": 0,
        "borderRight": 0,
        "borderTop": 0,
        "color": "#000000a6",
        "padding": "15px",
      }
    }
  >
    <a
      href="http://youtube.com"
      style={
        Object {
          "color": "#000000a6",
          "fontSize": 18,
          "fontWeight": "bold",
          "textDecoration": "none",
        }
      }
    >
      RSS Item title
    </a>
    <div
      style={
        Object {
          "margin": 0,
        }
      }
    >
      <p
        style={
          Object {
            "display": "inline-block",
            "fontSize": 12,
            "margin": 0,
          }
        }
      >
        2019-2-24
      </p>
    </div>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "I am text content",
        }
      }
      style={
        Object {
          "fontSize": 12,
          "margin": "1em 0",
        }
      }
    />
  </div>
</div>
`;

exports[`Feed List component should render items not found message when no items are passed  1`] = `
<div
  style={
    Object {
      "border": "1px solid #ebedf0",
      "borderRadius": 2,
      "padding": "20px",
      "position": "relative",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "border": "1px solid #ebedf0",
        "borderLeft": 0,
        "borderRight": 0,
        "borderTop": 0,
      }
    }
  >
    <h2>
      My feed title
    </h2>
  </div>
  <div
    style={Object {}}
  >
    No Items found
  </div>
</div>
`;
